public/app/features/alerting/unified/components/rule-editor/dag.ts

```diff
@@ -25,17 +28,38 @@ export function createDagFromQueries(queries: AlertQuery[]): Graph {
     const targets = getTargets(query.model);
 
     targets.forEach((target) => {
-      const isSelf = source === target;
-
-      if (source && target && !isSelf) {
-        graph.link(target, source);
+      if (source && target) {
```

Comment:
Since we're no longer checking for `isSelf` we now also get error for cyclical references to self

Reply:
Yep, will add this!